opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -124,48 +125,133 @@ message MetricDescriptor {
-    MONOTONIC_INT64 = 2;
+    // Determines if the points are Int64DataPoint or DoubleDataPoint, as well
+    // as the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
+
+    // If "true" means that the sum is monotonic.
+    bool is_monotonic = 3;
+
+    // TODO: Decide if knowing that the aggregated measurements are raw
+    // measurements is important or not. Equivalent of "Snapshot" in #162.
+  }
 
-    // DOUBLE values are double-precision floating-point numbers.
+  // Represents the type of a metric that is calculated by aggregating as a
+  // Histogram of all reported measurements over a time interval.
+  //
+  // A Metric of this Type MUST store its values as HistogramDataPoint.
+  message Histogram {
+    // It describes the value type of the measurement used to build this
+    // aggregation.
     //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    DOUBLE = 3;
+    // Determines the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
 
-    // MONOTONIC_DOUBLE values are monotonically increasing double-precision
-    // floating-point numbers.
+    // TODO: Decide if knowing that the aggregated measurements are raw
+    // measurements is important or not. Equivalent of "Snapshot" in #162.
+
+    // TODO: Decide if knowing that the Sum is monotonic is important or not,
+    // may be derived from the buckets definition.
+  }
+
+  // Represents the type of a metric that is calculated by computing a summary
+  // of all reported measurements over a time interval.
+  //
+  // A Metric of this Type MUST store its values as SummaryDataPoint.
+  message Summary {
+    // It describes the value type of the measurement used to build this
+    // aggregation.
     //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    MONOTONIC_DOUBLE = 4;
-
-    // Histogram measurement.
-    // Corresponding values are stored in HistogramDataPoint.
-    HISTOGRAM = 5;
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable quantiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Determines the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
+
+    // TODO: Decide if knowing that the aggregated measurements are raw
+    // measurements is important or not. Equivalent of "Snapshot" in #162.
```

Comment:
I realize now the source of confusion for me w.r.t. Temporality is that it sometimes applies to the individual Metric (i.e., is this a change in a sum or a total sum?) and it sometimes applies to the collection strategy (i.e., am I returning all "active" Metrics since the beginning of the process or since the last report).

I think we should find a way to distinguish these two ideas with terminology. A practical way to probe this issue is that I might like to encode the following things:

1. A Histogram of deltas from the start of the process (dimensionally reduced or not)
2. A Histogram of cumulatives from the start of the process (probably dimensionally reduced otherwise just one value)
3. A Histogram of deltas from a recent interval (dimensionally reduced or not)
4. A Histogram of cumulatives at the end of a recent interval (probably dimensionally reduced otherwise just one value)

There's a question as to whether (2) and (4) are any different, probably they are not, but you can imagine different interpretations because in (2) you can't forget any timeseries, and in (4) you can omit timeseries that hvaen't been "active" in the latest window. 

So, two ways to interpret Cumulative.

OTOH, I think we're having trouble distinguishing (1) and (2).

Reply:
Created https://github.com/open-telemetry/opentelemetry-proto/issues/184 to track this. I want us to make progress with the overall model.